python/lsst/pipe/tasks/photoCal.py

```diff
@@ -296,186 +251,28 @@ def DebugInfo(name):
     def __init__(self, refObjLoader, schema=None, **kwds):
         """!Create the photometric calibration task.  See PhotoCalTask.init for documentation
         """
-        RefMatchTask.__init__(self, refObjLoader, schema=None, **kwds)
+        pipeBase.Task.__init__(self, **kwds)
         self.scatterPlot = None
         self.fig = None
         if schema is not None:
-            self.usedKey = schema.addField("calib_photometryUsed", type="Flag",
+            self.usedKey = schema.addField("calib_photometry_used", type="Flag",
                                            doc="set if source was used in photometric calibration")
-            self.candidateKey = schema.addField("calib_photometryCandidate", type="Flag",
-                                                doc="set if source was a candidate for use in calibration")
-            self.reservedKey = schema.addField("calib_photometryReserved", type="Flag",
-                                               doc="set if source was reserved, so not used in calibration")
         else:
             self.usedKey = None
-            self.candidateKey = None
-            self.reservedKey = None
+        self.makeSubtask("match", refObjLoader=refObjLoader)
+        self.makeSubtask("reserve", columnName="calib_photometry", schema=schema,
+                         doc="set if source was reserved from photometric calibration")
```

Comment:
Why does this not match the docstring for the Field above? And why is there no doc for `"match"`?

Reply:
`doc` is required for the `ReserveSourcesTask`, just as `refObjLoader` is required for the `DirectMatchTask`.